app/models/repo.rb

```diff
@@ -149,6 +149,22 @@ def git_fetch
     end
   end
 
+  def create_branch(branch_name)
+    b = branches.new(
+      :name => branch_name,
+      :commit_uri => "https://github.com/#{branch_name}/commit/$commit",
+    )
+
+    # Make sure the branch is a real git branch before continuing and saving a record
+    return if !b.git_service.send(:rugged_repo).branches.exists?("origin/#{b.name}")
+
+    b.merge_target = "master"  # I know this shouldn't be, but I need it for the last_commit
+    b.last_commit  = b.git_service.merge_base
+    b.merge_target = nil
```

Comment:
We have a similar method that creates the master branch in `Repo.create_from_github!`...can we DRY those up?